jsapp/js/components/modalForms/formMedia.es6

```diff
@@ -0,0 +1,277 @@
+    actions.media.uploadMedia.completed.listen(this.onUploadCompleted);
+    actions.media.uploadMedia.failed.listen(this.onUploadFailed);
+    actions.media.deleteMedia.completed.listen(this.onGetMediaCompleted);
+    actions.media.deleteMedia.failed.listen(this.onDeleteMediaFailed);
+
+    actions.media.loadMedia(this.props.asset.uid);
+  }
+
+  /*
+   * action listeners
+   */
+
+  onGetMediaCompleted(uploadedAssets) {
+    this.setState({
+      uploadedAssets: uploadedAssets.results,
+      isUploadFilePending: false,
+      isUploadURLPending: false,
+      isVirgin: false,
+    });
+  }
+
+  onUploadFailed(response) {
+    this.setState({
+      fieldsErrors: response.responseJSON,
+      isUploadFilePending: false,
+      isUploadURLPending: false
+    });
+  }
+
+  /*
+   * Utilities
+   */
+
+  toBase64(file) {
+    return new Promise(function(resolve, reject) {
+      var reader = new FileReader();
+      reader.readAsDataURL(file);
+      reader.onload = function() {
+          return resolve(reader.result);
+      };
+      reader.onerror = function(error) {
+          return reject(error);
+      };
+    });
+  }
+
+  uploadMedia(formMediaJSON) {
+    // Reset error message before uploading again
+    this.setState({fieldsErrors: {}});
+    actions.media.uploadMedia(this.props.asset.uid, formMediaJSON);
+  }
+
+  /*
+   * DOM listeners
+   */
+
+  onFileDrop(files) {
+    if (files.length >= 1) {
+      this.setState({isUploadFilePending: true});
+
+      files.forEach(async (file) => {
+        var base64File = await this.toBase64(file);
+	    var formMediaJSON = {
+	      description: 'default',
+          file_type: 'form_media',
+          metadata: JSON.stringify({filename: file.name}),
+	      base64Encoded: base64File
+	    };
+        this.uploadMedia(formMediaJSON);
+      });
+    }
+  }
+
+  onInputURLChange(inputURL) {
+    this.setState({inputURL: inputURL});
+  }
+
+  onSubmitURL() {
+    var url = this.state.inputURL;
+
+    if (url === '') {
+      alertify.warning(t('URL is empty!'));
+    } else {
+      this.setState({
+        isUploadURLPending: true,
+        inputURL: ''
+      });
+
+      var formMediaJSON = {
+        description: 'default',
+        file_type: 'form_media',
+        metadata: JSON.stringify({redirect_url: url}),
+      };
+      this.uploadMedia(formMediaJSON);
+    }
+  }
+
+  onDeleteMedia(url) {
+    actions.media.deleteMedia(this.props.asset.uid, url);
+  }
+
+  uploadedAssetsIsEmpty() {
+    return (
+      this.state.uploadedAssets === null ||
+      this.state.uploadedAssets === undefined
+    );
+  }
+
+  /*
+   * rendering
+   */
+
+  renderButton() {
+    const buttonClassNames = ['formBuilder-header__button', 'formBuilder-header__button--save'];
+    if (this.state.isUploadURLPending) {
+      buttonClassNames.push('formBuilder-header__button--savepending');
+    }
+    return (
+      <button className={buttonClassNames.join(' ')}>
+        <i />
+        {t('ADD')}
```

Comment:
When I paste this: `https://upload.wikimedia.org/wikipedia/commons/2/28/Leszek_Kolakowski_1971.jpg` into the input and click "ADD" nothing happens.

Reply:
Hmm.. it works for me, you may have ran into a migration issue that we're trying to fix now

Reply:
I re-tested it and now can't reproduce it. It might be some problem on my part :)